src/schematools/types.py

```diff
@@ -377,18 +478,25 @@ def fields(self) -> Iterator[DatasetFieldSchema]:
         if self.has_compound_key:
             yield DatasetFieldSchema(_id="id", _parent_table=self, _required=True, type="string")
 
-    def get_fields_by_id(self, field_ids: List[str]) -> Iterator[DatasetFieldSchema]:
-        for field in self.fields:
-            if field.id in set(field_ids):
-                yield field
+    @lru_cache()
+    def get_fields_by_id(self, field_ids: Tuple[str]) -> Iterator[DatasetFieldSchema]:
```

Comment:
suggestie: `*field_ids: str` er van maken. Dan kan je ook overal de `tuple()` cast weglaten.

Reply:
If you decide to keep it as a tuple, you should probably change the type annotation to `Tupel[str, ...]`. As it is right now it will only accept one-element tuples.

Reply:
Goeie suggestie. Was eerst list, maar moest immutable worden. Er `args` van maken, maakt het nog iets beter leesbaar.